src/Data/Auth/AuthProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { AuthConfig, LocalConfig } from "./PrimaryAuthController";
+import { DatabaseAuthProvider } from "./Providers/DatabaseAuthProvider";
+import { KeycloakAuthProvider } from "./Providers/KeycloakAuthProvider";
+import { NoAuthProvider } from "./Providers/NoAuthProvider";
+
+interface Props {
+  config: undefined | AuthConfig | LocalConfig;
+  keycloakUrl: string | undefined;
+}
+
+/**
+ * AuthProvider component renders the appropriate authentication provider based on the provided config
+ *
+ * @param config - The authentication configuration.
```

Comment:
This "interface" should be 100% generic. I see config which makes me assume that this is the configuration provided by the backend but then there is still a keycloak specific parameter. This should not be the case:

1. It should be clear what `config` is and I cannot deduce that from the code, the naming or the comments
2. We should not have any authentication method specific parameters in here anymore.

Reply:
changed, and removed url parameter